### DISCUSSION TITLE
Using .clean() to call methods outside of spec

### DISCUSSION THREAD

**Discussion by treyturner** [pactumjs/pactum#273] [Q&A]
Hi there! I'm investigating Pactum for possible implementation in my organization. Is it possible, or would it be a valid feature request, to call methods outside of the `spec` for `cleanup()` purposes? Where possible we prefer to cleanup via the database so as to avoid putting extra API endpoints under test.

**Comment by ASaiAnudeep** [comment]
Hello @treyturner, if you are referring to `cleanup` method under [e2e testing](https://pactumjs.github.io/guides/e2e-testing.html), it's not possible to call it outside a single test. To better understand the feature you are requesting, it would be helpful if you could provide an example illustrating how it should be implemented and its expected functionality.

**Reply by treyturner** [reply]
Thanks for your response. What we'd be looking to do is to call one or more external async methods as part of the `.cleanup()` process. The goal in doing so would be to avoid a pattern of wrapping every test in a try/finally for cleanup purposes.

This is a contrived example largely copy/pasted from your docs, but essentially we'd be looking for something like
```
await spec()
    .post('http://jsonplaceholder.typicode.com/posts')
    .withJson({
        title: 'foo',
        body: 'bar',
        userId: 1,
    })
    .expectStatus(200)
    .stores('postId', '[0].id')
    .cleanup(async () => {
        await db.deletePost('$S{postId}');
    })
```

Does that make sense?

**Reply by gr00nd** [reply]
My apologies, 
I do not know how am I supposed to respond to this or if I should do anything? Please 
advise. I'm an artist and I am very interested in programming and studying and practicing it independently. 
I can be reached via email: 

Pax@gr00nd.com

Thank you again.
